src/main/java/wolox/training/models/Book.java

```diff
@@ -122,43 +122,75 @@ public String getIsbn() {
-        this.genre = checkNotNull(genre);
+        Preconditions
+            .checkNotNull(genre, "The genre field cannot be null.");
+        Preconditions
+            .checkArgument(! genre.isEmpty(), "The genre field cannot be empty.");
+        this.genre = genre;
     }
 
     private void setAuthor(String author) {
-        this.author = checkNotNull(author);
+        Preconditions
+            .checkNotNull(author, "The author field cannot be null.");
+        Preconditions
+            .checkArgument(! author.isEmpty(), "The author field cannot be empty.");
+        this.author = author;
     }
 
     private void setImage(String image) {
-        this.image = checkNotNull(image);
+        Preconditions
+            .checkNotNull(image, "The image field cannot be null.");
+        Preconditions
+            .checkArgument(! image.isEmpty(), "The image field cannot be empty.");
+        this.image = image;
     }
 
     private void setTitle(String title) {
-        this.title = checkNotNull(title);
+        Preconditions
+            .checkNotNull(title, "The title field cannot be null.");
+        Preconditions
+            .checkArgument(! title.isEmpty(), "The title field cannot be empty.");
+        this.title = title;
     }
 
     private void setSubtitle(String subtitle) {
-        this.subtitle = checkNotNull(subtitle);
+        Preconditions
+            .checkNotNull(subtitle, "The subtitle field cannot be null.");
+        Preconditions
+            .checkArgument(! subtitle.isEmpty(), "The subtitle field cannot be empty.");
+        this.subtitle = subtitle;
     }
 
     private void setPublisher(String publisher) {
-        this.publisher = checkNotNull(publisher);
+        Preconditions
+            .checkNotNull(publisher, "The publisher field cannot be null.");
+        Preconditions
+            .checkArgument(! publisher.isEmpty(), "The publisher field cannot be empty.");
+        this.publisher = publisher;
     }
 
     private void setYear(String year) {
-        this.year = checkNotNull(year);
+        Preconditions
+            .checkNotNull(year, "The year field cannot be null.");
+        Preconditions
+            .checkArgument(! year.isEmpty(), "The year field cannot be empty.");
+        this.year = year;
     }
 
     private void setPage(Integer page) {
-        this.page = checkNotNull(page);
+        Preconditions
+            .checkNotNull(page, "The author field cannot be null.");
+        Preconditions
+            .checkArgument(! (page > 0), "The author field cannot be empty.");
+        this.page = page;
     }
 
     private void setIsbn(String isbn) {
-        this.isbn = checkNotNull(isbn);
-    }
-
-    private void setUsers(List<User> list) {
-        this.users = list;
+        Preconditions
+            .checkNotNull(isbn, "The isbn field cannot be null.");
+        Preconditions
+            .checkArgument(! isbn.isEmpty(), "The isbn field cannot be empty.");
```

Comment:
Check if the entered is a number instead of if it is empty:

```suggestion
            .checkArgument(StringUtils.isNumeric(isbn), "error message");
```